android/src/main/java/com/reactlibrary/RNSpotifyRemoteAppModule.java

```diff
@@ -66,52 +77,76 @@ public void onFailure(Throwable throwable) {
                     } else if (throwable instanceof CouldNotFindSpotifyApp) {
                         promise.reject(new Error("Spotify connection failed: could not find the Spotify app, it may need to be installed."));
                     } else {
-                        promise.reject(throwable);   
+                        promise.reject(throwable); 
                     }
                 }
-                sendEvent("remoteDisconnected", null);
+                sendEvent(reactContext, EventNameRemoteDisconnected, null);
             }
         };
     }
 
-    @Override
-    @ReactMethod
-    public void __registerAsJSEventEmitter(int moduleId) {
-        RNEventEmitter.registerEventEmitterModule(this.reactContext, moduleId, this);
-    }
-
-    @Override
-    public void onNativeEvent(String eventName, Object... args) {
-        // Called when an event for this module is emitted from native code
+    private void sendEvent(ReactApplicationContext reactContext,
```

Comment:
Since this is private, can we just use the private `reactContext` instead of passing it in?  Or does the context ever change?

Reply:
Yeah, we can just reuse it actually. I just follow the how the React Native doc recommends writing these functions. Will make a commit when I am available .